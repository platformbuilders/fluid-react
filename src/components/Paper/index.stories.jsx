import React from 'react';
import { storiesOf } from '@storybook/react';
import markdown from './paper.md';
import Paper from './index';

storiesOf('Paper', module)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Paper>
      <h4>Simple Wrapper</h4>
    </Paper>
  ));
