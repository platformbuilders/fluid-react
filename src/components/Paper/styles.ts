import styled from 'styled-components';
import { Paper } from '@material-ui/core';
import { getTheme } from '../../utils/helpers';

export const Wrapper = styled(Paper)`
  padding: ${getTheme('largeSpacing')}px;
`;
