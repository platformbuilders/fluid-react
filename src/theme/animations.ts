import LoadingAnimation from '../assets/animations/loading.json';
import { Animations } from '../types';

export default {
  circularLoading: LoadingAnimation,
  buttonLoading: LoadingAnimation,
  linearLoading: LoadingAnimation,
} as Animations;
