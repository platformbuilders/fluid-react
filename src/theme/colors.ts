import { lighten, darken } from 'polished';
import { Colors } from '../utils/types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';

const colors: Colors = {
  failure: '#cc0000',
  warning: '#F5B800',
  success: '#357a38',
  transparent: 'transparent',
  disabled: '#EBEBE4',
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
