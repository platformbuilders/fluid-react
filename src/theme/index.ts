export { default as colors } from './colors';
export { default as radius } from './radius';
export { default as animations } from './animations';
