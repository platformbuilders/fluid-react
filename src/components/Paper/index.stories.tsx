import type { Meta, StoryObj } from '@storybook/react';
import Paper from './index';

const meta: Meta<typeof Paper> = {
  title: 'Paper',
  component: Paper,
  args: {
    children: <h4>teste</h4>,
  },
};

export default meta;
type Story = StoryObj<typeof Paper>;

export const Default: Story = {};
