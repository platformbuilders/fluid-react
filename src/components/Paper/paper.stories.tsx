import { expect } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, within } from '@storybook/testing-library';
import Paper from './index';

configure({ testIdAttribute: 'id' });

//  Mocks
const mockTextId = 'paper-id';

const meta: Meta<typeof Paper> = {
  title: 'Components/Paper',
  component: Paper,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {},
  args: {
    children: <h3>Test</h3>,
    elevation: 'z5',
    id: mockTextId,
    square: true,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Paper | Test Render', async () => {
      expect(canvas.getByTestId(mockTextId)).toBeInTheDocument();
    });
  },
};

type Story = StoryObj<typeof Paper>;

export const Default: Story = {
  args: {},
};

export default meta;
