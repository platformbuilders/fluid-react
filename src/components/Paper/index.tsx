import React, { FC } from 'react';
import { Wrapper } from './styles';

type Props = {
  className?: string;
  children?: React.ReactNode;
};

const Paper: FC<Props> = ({ className, children, ...rest }): JSX.Element => (
  <Wrapper className={className} {...rest}>
    {children}
  </Wrapper>
);

export default Paper;
