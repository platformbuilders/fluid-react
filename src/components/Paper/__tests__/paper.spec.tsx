import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { faker } from '@faker-js/faker';

import Paper from '..';
import theme from '../../../theme';

const defaultContent = 'Text';
const defaultProps = {};

describe('Component: Paper', () => {
  test('snapshots', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Paper {...defaultProps}>{defaultContent}</Paper>
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('check props', () => {
    // when
    const mockContent = faker.random.word();
    const component = shallow(<Paper {...defaultProps}>{mockContent}</Paper>);

    // then
    expect(component.children().text()).toEqual(mockContent);
  });
});
