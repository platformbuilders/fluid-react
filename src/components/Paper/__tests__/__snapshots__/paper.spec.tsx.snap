// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Paper snapshots 1`] = `
<div>
  <div
    class="sc-bcHOTY cCjbEw"
  >
    Text
  </div>
</div>
`;
